annotation/tests/test_annotations.py

```diff
@@ -83,3 +172,142 @@ def test_row_to_dict(row, expected_dictionary):
     result = row_to_dict(row)
     print(result)
     assert result == expected_dictionary
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_prefix",
+        "bucket_name",
+        "expected_string",
+    ],
+    [
+        (
+            "S3_test",
+            "bucket_test",
+            "S3_test-bucket_test",
+        ),
+        (
+            None,
+            "bucket_test",
+            "bucket_test",
+        ),
+    ],
+)
+def test_convert_bucket_name_if_s3prefix(
+    monkeypatch,
+    s3_prefix,
+    bucket_name,
+    expected_string,
+):
+    monkeypatch.setattr("annotation.annotations.main.S3_PREFIX", s3_prefix)
+    result = convert_bucket_name_if_s3prefix(bucket_name=bucket_name)
+    assert result == expected_string
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_provider",
+        "bucket_name",
+    ],
+    [
+        (
+            "minio",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "aws_iam",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "no_provider",
+            TEST_TENANT,
+        ),  # NotConfiguredException
+        (
+            "aws_iam",
+            "no_bucket",
+        ),  # NoSuchBucket
+    ],
+)
+def test_connect_s3(moto_s3, monkeypatch, s3_provider, bucket_name):
+    mock_resource = MagicMock(return_value=moto_s3)
+    monkeypatch.setattr("boto3.resource", mock_resource)
+
+    monkeypatch.setattr("annotation.annotations.main.S3_PROVIDER", s3_provider)
+
+    if s3_provider == "no_provider":
+        with pytest.raises(NotConfiguredException):
+            connect_s3(bucket_name)
+    elif bucket_name == "no_bucket":
+        with pytest.raises(moto_s3.meta.client.exceptions.NoSuchBucket):
+            connect_s3(bucket_name)
+    else:
+        result_s3 = connect_s3(bucket_name)
+        assert moto_s3 == result_s3
+
+
+@pytest.mark.unittest
```

Comment:
dont add this decorator in any test, wont need em

Reply:
Noted, removed from tests.